cfme/tests/infrastructure/test_scvmm_specific.py

```diff
@@ -30,6 +47,99 @@ def vm(provider, small_template):
     vm.cleanup_on_provider()
 
 
+@pytest.fixture
+def cfme_vhd(provider, appliance):
+    """ Given a stream from the appliance, gets the cfme vhd"""
+    stream = appliance.version.stream()
+    try:
+        url = '{}/'.format(conf.cfme_data["basic_info"]["cfme_images_url"][stream])
+    except KeyError:
+        pytest.skip("No such stream: {} found in cfme_data.yaml".format(stream))
+    # get image name
+    image_name = get_vhd_name(url)
+    if not image_name:
+        pytest.skip("No hyperv vhd image at {}".format(url))
+    # download the image to SCVMM library
+    provider.mgmt.download_file(url + image_name, image_name)
+    provider.mgmt.update_scvmm_library()
+    # check to make sure that the file is in the library
+    assert image_name in provider.mgmt.list_vhds()
+
+    yield image_name
```

Comment:
Q: do you need to remove the image after the test?

Reply:
oh, I see it's removed after provisioning. Maybe worth moving it here in case of some pytest.skip in the provisioning

Reply:
It needs to be deleted last, because you can't delete it if a vm or template is using that vhd image. That's why I do the cleanup in `scvmm_appliance`. Do you know if the cleanup in `cfme_vhd` will execute before the cleanup in `scvmm_appliance`? Or in general a way to control fixture cleanup execution?

Reply:
@john-dupuy ah, interesting. Looks like it should work correctly be design https://github.com/pytest-dev/pytest/issues/1581